r/configure

```diff
@@ -298,6 +254,11 @@ set_pkg_vars () {
   if [ "$ARROW_R_CXXFLAGS" ]; then
     PKG_CFLAGS="$PKG_CFLAGS $ARROW_R_CXXFLAGS"
   fi
+
+  if [ "$UNAME" = "Darwin" ] && [ $(expr `sw_vers -productVersion` : '10\.13' ) -gt 3 ]; then
```

Comment:
Is this really equivalent? I think this would match e.g. 11 or 10.15 with atleast one char and return a truthy exit code?
>  If the match succeeds and the pattern contains at least one regu-
       lar expression subexpression \(...\), the string correspond-
       ing to \1 is returned; otherwise the matching operator
       returns the number of characters matched.  If the match fails and
       the pattern contains a regular expression subexpression the null
       string is returned; otherwise 0.

Reply:
Hmm. It seems working:

```cosnole
$ expr 11 : '10\.13' > /dev/null; echo $?
1
$ expr 10.15 : '10\.13' > /dev/null; echo $?
1
$ expr 10.13 : '10\.13' > /dev/null; echo $?
0
$ expr 10.13.9 : '10\.13' > /dev/null; echo $?
0
```

Reply:
Ok, just wanted to make sure as I don't work on a mac ^^